.github/workflows/ci.yaml

```diff
@@ -3,17 +3,63 @@ name: default
   push:
     branches:
       - main
+      - release-*
     tags:
       - v*
-  pull_request: {}
-env:
-  CI_ARGS: --cache-from=type=registry,ref=registry.dev.siderolabs.io/${GITHUB_REPOSITORY}:buildcache --cache-to=type=registry,ref=registry.dev.siderolabs.io/${GITHUB_REPOSITORY}:buildcache,mode=max
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+    branches:
+      - main
+      - release-*
+  schedule:
+    - cron: '30 1 * * 1'
 jobs:
   default:
+    if: ${{ !startsWith(github.head_ref, 'renovate/') }}
     permissions:
-      contents: write
       packages: write
-    runs-on: self-hosted
+    runs-on:
+      - self-hosted
+      - pkgs
+    services:
+      buildkitd:
+        image: moby/buildkit:buildx-stable-1
+        ports:
+          - 1234:1234
+        options: --privileged
+        volumes:
+          - /var/lib/buildkit/${{ github.repository }}:/var/lib/buildkit
+          - /usr/etc/buildkit/buildkitd.toml:/etc/buildkit/buildkitd.toml
     steps:
       - name: checkout
         uses: actions/checkout@v3
+      - name: Unshallow
+        run: |
+          git fetch --prune --unshallow
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          driver: remote
+          endpoint: tcp://localhost:1234
+          append: |
+            - endpoint: tcp://buildkit-arm64.ci.svc.cluster.local:1234
+              platforms: linux/arm64
+      - name: build
+        if: github.event_name != 'schedule'
+        run: |
+          make
+      - name: reproducibility
```

Comment:
no, only on schedule or when there's the label

Reply:
yep, but should it be a separate workflow? would it be cleaner?

this will have its own status, as `reproducibility` is independent of all other steps